services/device-connection/src/main/java/org/eclipse/hono/deviceconnection/infinispan/RemoteCacheBasedDeviceConnectionService.java

```diff
@@ -86,28 +84,36 @@ protected final void doStart(final Promise<Void> startPromise) {
     @Override
     protected void doStart(final Future<Void> startFuture) {
 
-        final Promise<Void> result = Promise.promise();
-        result.future().setHandler(startFuture);
-
         if (cacheManager == null) {
-            result.fail(new IllegalStateException("cache manager is not set"));
+            startFuture.fail(new IllegalStateException("cache manager is not set"));
         } else {
-            context.executeBlocking(r -> {
-                cacheManager.start();
-                cache = cacheManager.getCache("device-connection");
-                cache.start();
-                r.complete(cacheManager);
-            }, attempt -> {
-                if (attempt.succeeded()) {
-                    LOG.info("successfully connected to remote cache");
-                } else {
-                    LOG.info("failed to connect to remote cache", attempt.cause());
-                }
-            });
-            result.complete();
+            connectToGrid();
+            startFuture.complete();
         }
     }
 
+    private void connectToGrid() {
+
+        context.executeBlocking(r -> {
```

Comment:
I think the block misses the error handling:

> A Future instance is passed into blockingCodeHandler. When the blocking code successfully completes, the handler should call the Promise.complete(T) or Promise.complete(Object) method, or the Promise.fail(java.lang.Throwable) method if it failed.

Reply:
`ContextImpl.executeBlocking` already takes care of it by failing the promise if the blocking code throws a `Throwable`. However, this behavior is not specified by `Context.executeBlocking()`. Given that the infinispan methods in question do not document any exceptions being thrown, we probably should explicitly catch all `Throwable`s here for transparency ...